red-hat-data-services/odh-model-controller
# Oneroute

## Description
This PR modifies odh-model-controller as follows : 

- 1 route will be created PER namespace for accessing all models in the namespace. 
- Route creation/deletion is controller by creation/deletion of ServingRuntimes instead of InferenceServices
- Old routes will automatically not be deleted if they were created BEFORE the upgrade (one routes being the ones created per inferenceservice)
- Old route will be deleted if the associated inferenceservice is deleted (same behavior as before)

## Testing
- Deploy existing production RHODS
- Deploy a model server and isvc and confirm 1 route is created for the isvc
- Uninstall RHODS (do not delete the model server and isvc) 
- Install rhods using http://quay.io/vedantm/rhods-operator-live-catalog:1.28.0-oneroute 
- Navigate to the NS with the model server
- Verify 2 routes exist (old route + new <namespace>-model-route) 
- Create a NEW isvc with a different name, verify this new isvc does not get it's own route 
- Delete original isvc, verify if the route associated with the isvc got deleted but the new <namespace>-model-route still exists. 
- Delete model server and verify the associated route is deleted. 

JIRA: https://issues.redhat.com/browse/RHODS-5888 
Doc: https://docs.google.com/document/d/1PCL3wgkk_OSY9Km9vOt2kZ4t6lEtitj223dYea-Uuk4/edit#heading=h.w5e37xee1yqq
ADR : TBD
